### PR TITLE
Install `twine` in `cibuildwheel` environment

### PR DESCRIPTION
**Pull request by charlesbluca**
It looks like the `twine` installed prior to running `cibuildwheel` is not picked up in the Mac builds, potentially due to the venv that `cibuildwheel` creates.

This PR attempts to resolve this by installing `twine` as part of the `cibuildwheel` builds.

### CODE REVIEW

**Comment by charlesbluca** [comment]
Looks like even with the Python setup action, `twine` isn't getting installed properly on macOS - going to try using conda instead of pip to see if that resolves

File: .github/workflows/release.yml
Comment (ayushdg) [inline .github/workflows/release.yml:5]: Commenting as a reminder to remove before merging

**Review by ayushdg — APPROVED** [review]
Changes lgtm! Should be good to merge when tests pass (after removing the pr trigger)

**Pull request merged**
